Assets/Scripts/MovePlayer.cs

```diff
@@ -0,0 +1,213 @@
+    /// </summary>
+    private directions currentDirection;
+    /// <summary>
+    /// An array for quick reference of directions, in conjunction with the enum
+    /// This prevents the need to write four separate coroutines
+    /// </summary>
+    private Vector3[] movementVectors;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        playerAnimator = GetComponent<Animator>();
+        movementVectors = new Vector3[4] { Vector2.down, Vector2.up, Vector2.left, Vector2.right };
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        GetInput();
+    }
+
+    private void FixedUpdate()
+    {
+        if (canMove && shouldMove && !moving)
+            StartCoroutine(MovementRoutine());
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.gameObject.tag == slowTerrainTag)
+            slowMove = !slowMove;
+    }
+
+    /// <summary>
+    /// Gets player input
+    /// </summary>
+    private void GetInput()
+    {
+        if (Input.GetAxis("Vertical") < 0)
+        {
+            shouldMove = true;
+            currentDirection = directions.Down;
+        }
+        else if (Input.GetAxis("Vertical") > 0)
+        {
+            shouldMove = true;
+            currentDirection = directions.Up;
+        }
+        else if (Input.GetAxis("Horizontal") < 0)
+        {
+            shouldMove = true;
+            currentDirection = directions.Left;
+        }
+        else if (Input.GetAxis("Horizontal") > 0)
+        {
+            shouldMove = true;
+            currentDirection = directions.Right;
+        }
+        else
+            shouldMove = false;
+    }
+
+    /// <summary>
+    /// Adjusts the player to fit exactly into a tile
+    /// </summary>
+    private void SnapToGrid()
+    {
+        int positionX = Mathf.RoundToInt(transform.position.x);
+        int positionY = Mathf.RoundToInt(transform.position.y);
+        transform.position = new Vector3(positionX, positionY);
+    }
+
+
+    private void PlayWalkAnim(directions movementDirection, bool slowTerrain)
+    {
+        playerAnimator.SetBool("Slow", slowTerrain);
+        switch (movementDirection)
+        {
+            case directions.Down:
+                playerAnimator.SetTrigger("Move Down");
+                break;
+            case directions.Up:
+                playerAnimator.SetTrigger("Move Up");
+                break;
+            case directions.Left:
+                playerAnimator.SetTrigger("Move Left");
+                break;
+            case directions.Right:
+                playerAnimator.SetTrigger("Move Right");
+                break;
+        }
+    }
+
+    /// <summary>
+    /// Raycasts for collision in the current direction before moving
+    /// </summary>
+    /// <returns>True if there is something in the way, else false.</returns>
+    private bool CheckCollisionRaycast(directions attemptedMoveDirection)
```

Comment:
Bool return type method should be named as a yes or no question. IsCollisionBlockingMovement or something like that.